packages/instant-apps-components/src/components/instant-apps-export/instant-apps-export.tsx

```diff
@@ -611,4 +631,152 @@ export class InstantAppsExport {
       }
     }
   }
+
+  createScreenshot(): void {
+    if (this.view != null) {
+      this.screenshotPreview = document.createElement('div');
+      this.screenshotPreview.className = 'screenshot-preview hide';
+      this.screenshotImgContainer = document.createElement('div');
+      this.screenshotImgContainer.className = 'screenshot-img-container';
+      this.screenshotImg = document.createElement('img');
+      const screenshotBtnContainer = document.createElement('div');
+      const exportBtn = document.createElement('calcite-button');
+      const returnBtn = document.createElement('calcite-button');
+      exportBtn.innerHTML = this.messages?.export;
+      returnBtn.innerHTML = this.messages?.returnToMap;
+      returnBtn.appearance = 'outline-fill';
+      exportBtn.onclick = this.exportOnClick.bind(this);
+      returnBtn.onclick = this.screenshotReturn.bind(this);
+      screenshotBtnContainer.append(returnBtn, exportBtn);
+      this.screenshotImgContainer.append(this.screenshotImg, screenshotBtnContainer);
+      this.screenshotPreview.append(this.screenshotImgContainer);
+      this.view.container.append(this.screenshotPreview);
```

Comment:
looks like these nodes are created with the document API and appended to the map view container. Any reason why JSX isn't being used?

Reply:
Because it needs to be appended to the view. JSX gets compiled to javascript anyway.